src/coldfront_plugin_api/scim_v2/users.py

```diff
@@ -0,0 +1,112 @@
+from django.contrib.auth.models import User
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAdminUser
+from rest_framework.response import Response
+
+from coldfront_plugin_api import auth, utils
+
+
+def user_to_api_representation(user: User) -> dict:
+    return {
+        "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
+        "id": user.username,
+        "externalId": user.username,
+        "userName": user.username,
+        "name": {
+            "givenName": user.first_name,
+            "familyName": user.last_name,
+        },
+        "emails": [
+            {
+                "value": user.email,
+                "type": "work",
+                "primary": True,
+            }
+        ],
+        "meta": {
+            "resourceType": "User",
+            "created": user.date_joined,
+        },
+    }
+
+
+class ListUsers(APIView):
+    """
+    View to list all users in the system.
+
+    * Requires token authentication.
+    * Only admin users are able to access this view.
+    """
+    authentication_classes = auth.AUTHENTICATION_CLASSES
+    permission_classes = [IsAdminUser]
+
+    def get(self, request, format=None):
+        """
+        Return a list of all users.
+        """
+        users = User.objects.all()
+        return Response(
+            [user_to_api_representation(user) for user in users]
+        )
+
+    def post(self, request, format=None):
+        """
+        Create a new user.
+
+        At a minimum, payload needs to contain the following:
+        {
+            "schemas":["urn:ietf:params:scim:schemas:core:2.0:User"],
+            "userName":"fake-user-1"
+        }
+
+        If user exists through one of the configured search providers,
+        additional information will be fetched from there.
+
+        Otherwise, for setting the attributes manually, please see function
+        user_to_api_representation for example payload.
+        """
+        username = request.data.get("userName", "")
+
+        if (
+            request.data["schemas"] != ["urn:ietf:params:scim:schemas:core:2.0:User"]
```

Comment:
Should this by `request.data.get("schemas")`, or it is okay to raise a KeyError here?